tests/unit/test_diffsync_model_flags.py

```diff
@@ -0,0 +1,196 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import pytest
+
+from diffsync.enum import DiffSyncModelFlags
+from diffsync.exceptions import ObjectNotFound
+
+
+def test_diffsync_diff_with_skip_unmatched_src_flag_on_models(backend_a, backend_a_with_extra_models):
+    # Validate that there are 2 extras objects out of the box
+    diff = backend_a.diff_from(backend_a_with_extra_models)
+    assert diff.summary() == {"create": 2, "update": 0, "delete": 0, "no-change": 23}
+
+    # Check that only 1 object is affected by the flag
+    backend_a_with_extra_models.get(
+        backend_a_with_extra_models.site, "lax"
+    ).model_flags |= DiffSyncModelFlags.SKIP_UNMATCHED_SRC
+    diff = backend_a.diff_from(backend_a_with_extra_models)
+    assert diff.summary() == {"create": 1, "update": 0, "delete": 0, "no-change": 23}
+
+    # Check that only the parent object is affected by the flag not the children
+    backend_a_with_extra_models.get(
+        backend_a_with_extra_models.device, "nyc-spine3"
+    ).model_flags |= DiffSyncModelFlags.SKIP_UNMATCHED_SRC
+    diff = backend_a.diff_from(backend_a_with_extra_models)
+    assert diff.summary() == {"create": 0, "update": 0, "delete": 0, "no-change": 23}
+
+
+def test_diffsync_sync_with_skip_unmatched_src_flag_on_models(backend_a, backend_a_with_extra_models):
+    backend_a_with_extra_models.get(
+        backend_a_with_extra_models.site, "lax"
+    ).model_flags |= DiffSyncModelFlags.SKIP_UNMATCHED_SRC
+    backend_a_with_extra_models.get(
+        backend_a_with_extra_models.device, "nyc-spine3"
+    ).model_flags |= DiffSyncModelFlags.SKIP_UNMATCHED_SRC
+
+    backend_a.sync_from(backend_a_with_extra_models)
+
+    # New objects should not have been created
+    with pytest.raises(ObjectNotFound):
+        backend_a.get(backend_a.site, "lax")
+    with pytest.raises(ObjectNotFound):
+        backend_a.get(backend_a.device, "nyc-spine3")
+    assert "nyc-spine3" not in backend_a.get(backend_a.site, "nyc").devices
+
+
+def test_diffsync_diff_with_skip_unmatched_dst_flag_on_models(backend_a, backend_a_minus_some_models):
+    # Validate that there are 3 extras objects out of the box
+    diff = backend_a.diff_from(backend_a_minus_some_models)
+    assert diff.summary() == {"create": 0, "update": 0, "delete": 12, "no-change": 11}
+
+    # Check that only 1 object is affected by the flag
+    backend_a.get(backend_a.device, "rdu-spine1").model_flags |= DiffSyncModelFlags.SKIP_UNMATCHED_DST
+    diff = backend_a.diff_from(backend_a_minus_some_models)
+    assert diff.summary() == {"create": 0, "update": 0, "delete": 9, "no-change": 11}
+
+    # Check that only 1 object is affected by the flag
+    backend_a.get(backend_a.device, "sfo-spine2").model_flags |= DiffSyncModelFlags.SKIP_UNMATCHED_DST
+    diff = backend_a.diff_from(backend_a_minus_some_models)
+    assert diff.summary() == {"create": 0, "update": 0, "delete": 5, "no-change": 11}
```

Comment:
Comments each say "only 1 object" but it looks like the diffs have 3 objects "disappear" in the first case and four more objects "disappear" in the second case?

Reply:
This is because the interfaces of the device are also affected, I've updated the comment to be more explicit